PlayList.java

```diff
@@ -108,8 +146,19 @@ public void add(PlayList other) {
      *  If start is negative or greater than size - 1, returns -1.
      */
     private int minIndex(int start) {
-        //// replace the following statement with your code
-        return 0;
+        int shortestDurIndex = 0;
+        int index = start;
+        if (start == size - 1) return size - 1;
+        if (start < 0 || start >= size) return -1;
+        for (int trackNum = start; trackNum < size - 1; trackNum++) {
+            if (tracks[index].isShorterThan(tracks[trackNum + 1])) {
+                shortestDurIndex = index;
+            } else { 
+                index = trackNum + 1;
+                shortestDurIndex = index;
+            }
```

Comment:
I really struggle to understand what this section of code should do.
However, without this the method will work as needed.
```suggestion
```